### ISSUE TITLE
`bazel test ...` does not work

### ISSUE THREAD

**Issue by lberki** [bazelbuild/bazel#594]
There appear to be two separate issues:

1. ... happily recurses into the bazel-* symlinks
2. When bazel-repo/external is hit, a NullPointerException is raised

At least the first issue should be fixed.

**Comment by lberki** [comment]
paging also @kchodorow. 

Reproduction is:

```
git clone https://github.com/google/cctz.git
cd cctz
${BAZEL} test ...
${BAZEL} shutdown
#{BAZEL} test ...
The stack trace is as follows:


```
Unhandled exception thrown during build; message: Unrecoverable error while evaluating node 'FILE_STATE:[/]/[home/jeff/.cache/bazel/_bazel_jeff/95e8c9d3a6bb84d9bfbc625fe12d8152/external]' (requested by nodes 'FILE:[/tmp/google/cctz]/[bazel-cctz/external]')
INFO: Elapsed time: 0.328s
java.lang.RuntimeException: Unrecoverable error while evaluating node 'FILE_STATE:[/]/[home/jeff/.cache/bazel/_bazel_jeff/95e8c9d3a6bb84d9bfbc625fe12d8152/external]' (requested by nodes 'FILE:[/tmp/google/cctz]/[bazel-cctz/external]')
    at com.google.devtools.build.skyframe.ParallelEvaluator$Evaluate.run(ParallelEvaluator.java:974)
    at com.google.devtools.build.lib.concurrent.AbstractQueueVisitor$2.run(AbstractQueueVisitor.java:496)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
    at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.NullPointerException
    at com.google.common.base.Preconditions.checkNotNull(Preconditions.java:210)
    at com.google.devtools.build.lib.skyframe.ExternalFilesHelper.maybeHandleExternalFile(ExternalFilesHelper.java:136)
    at com.google.devtools.build.lib.skyframe.FileStateFunction.compute(FileStateFunction.java:47)
    at com.google.devtools.build.skyframe.ParallelEvaluator$Evaluate.run(ParallelEvaluator.java:938)
    ... 4 more
```

**Comment by schroederc** [comment]
Any progress on this?  It's P1, and it blocks Kythe's adoption of newer versions of Bazel.

**Comment by lberki** [comment]
This has been fixed in c509e33d4d2a648a1eb8dedb2212215ade35667a . Sorry for not closing this bug in time :(
